File: javargarten/src/irrgarten/Weapon.java
package irrgarten;


public class Weapon {
/*Esta clase representa las armas que utiliza el jugador en los ataques 
durante los combates.*/
    private float power;
    private int uses;
    
    /**************************************************************************/

    /*Constructor por defecto: 
    crea sin daño ni usos*/
    public Weapon(){this(0,0);}
    /*Constructor con especificación de cada parámetro */
    public Weapon(float power, int uses){
        this.power = power;
        this.uses = uses;
    }
    /**************************************************************************/




    /*Devuelve  la intensidad del ataque del jugador y resta un uso al arma 
    si ésta puede atacar (uses>0). */
    public float attack(){
        if (uses > 0){
            uses--;
            return power;
        }
        return 0;
    }

    /*Decide si un arma debe ser descartado con ayuda del Dado. */
    public boolean discard(){ return Dice.discardElement(uses); }

    //Devuelve una String del formato 'W[{power}, {uses}]' 
    @Override
    public String toString(){ return "W[" + power + ", " + uses + "]"; }
}

File: rubyrrgarten/Irrgarten/Weapon.rb
module Irrgarten
  require_relative 'Dice'
  class Weapon
    # Atributos
    attr_reader :power, :uses

    # Constructor
    def initialize(power_, uses_)
      @power = power_
      @uses = uses_
    end

    # Método para realizar un ataque
    def attack
      if @uses > 0
        @uses -= 1  # Decrementa el uso
        @power  # Devuelve el poder del ataque
      else
        0  # Devuelve 0 si no hay usos disponibles
      end
    end

    # Método para decidir si el arma debe ser descartada
    def discard
      Dice.discard_element(@uses)  # Llama al método de la clase Dice
    end

    # Método para representar el arma como un string
    def to_s
      "W[#{@power}, #{@uses}]"  # Formato 'W[{power}, {uses}]'
    end

  end
end
